### PR TITLE
HLRC: ML Update Job

### PR DESCRIPTION
**Pull request by benwtrent**
Adds the ability to update a Machine Learning job through the HLRC.

This relates to (#29827)

### CODE REVIEW

File: client/rest-high-level/src/main/java/org/elasticsearch/client/MachineLearningClient.java [hunk 1]
```diff
@@ -356,6 +358,38 @@ public void flushJobAsync(FlushJobRequest request, RequestOptions options, Actio
                 Collections.emptySet());
     }
 
+    /**
+     * Updates a Machine Learning {@link org.elasticsearch.client.ml.job.config.Job} properties
```
Comment (dimitris-athanasiou) [inline client/rest-high-level/src/main/java/org/elasticsearch/client/MachineLearningClient.java]: should we delete `properties`?

File: client/rest-high-level/src/main/java/org/elasticsearch/client/ml/UpdateJobRequest.java [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+import java.io.IOException;
+import java.util.Objects;
+
+/**
+ * Updates a {@link org.elasticsearch.client.ml.job.config.Job} with the passed {@link JobUpdate}
+ * settings
+ */
+public class UpdateJobRequest extends ActionRequest implements ToXContentObject {
+
+    private JobUpdate update;
```
Comment (dimitris-athanasiou) [inline client/rest-high-level/src/main/java/org/elasticsearch/client/ml/UpdateJobRequest.java]: make `final`

File: docs/java-rest/high-level/ml/update-job.asciidoc [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+=== Update Job API
+
+The Update Job API provides the ability to update a {ml} job.
+It accepts a `UpdateJobRequest` object and responds
+with a `PutJobResponse` object.
+
+[[java-rest-high-x-pack-ml-update-job-request]]
+==== Update Job Request
+
+A `UpdateJobRequest` object gets created with an `JobUpdate` object.
```
Comment (dimitris-athanasiou) [inline docs/java-rest/high-level/ml/update-job.asciidoc]: `an` -> `a`

**Review by dimitris-athanasiou — COMMENTED** [review]
Looks good! Left a few minor comments.

File: docs/java-rest/high-level/ml/update-job.asciidoc (hunk 3 above)
Comment (benwtrent) [reply docs/java-rest/high-level/ml/update-job.asciidoc]: I should just always make `The` so there is a certain authority and exactness to it :)

**Review by dimitris-athanasiou — APPROVED** [review]
LGTM

**Pull request merged**